client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts

```diff
@@ -54,23 +54,27 @@ const isWPCOM = ( siteInfo?: UrlData ) => {
 	return !! siteInfo?.platform_data?.is_wpcom;
 };
 
-const getFormDefaultValues = ( fromUrl: string ): CredentialsFormData => {
-	return {
-		from_url: fromUrl,
-		username: '',
-		password: '',
-		backupFileLocation: '',
-		notes: '',
-		migrationType: 'credentials',
-	};
+const removeEndingSlash = ( url: string ) => {
+	try {
+		const urlObject = new globalThis.URL( url );
+		if ( urlObject.pathname === '/' && url.endsWith( '/' ) ) {
+			return url.slice( 0, -1 );
+		}
+	} catch {
+		if ( url.endsWith( '/' ) ) {
+			return url.slice( 0, -1 );
+		}
```

Comment:
If it's not a valid URL, I think we should simply return it as-is, rather than trying to remove a trailing slash from some unexpected string.